include/camp/device.hpp

```diff
@@ -0,0 +1,313 @@
+#ifndef __CAMP_DEVICES_HPP
+#define __CAMP_DEVICES_HPP
+
+#include <cstring>
+#include <memory>
+
+#include <cuda_runtime.h>
+namespace camp
+{
+namespace devices
+{
+
+  enum class Platform {
+    undefined = 0,
+    host = 1,
+    omp = 2,
+    tbb = 4,
+    cuda = 8,
+    // omp_target = 16, not sure this is a meaningful difference
+    hip = 32
+  };
+
+  class CudaEvent
+  {
+    public:
+      CudaEvent(){ cudaEventCreateWithFlags(&m_event, cudaEventDisableTiming); }
+      void capture(cudaStream_t stream){ cudaEventRecord(m_event, stream); }
```

Comment:
I thought we were going to drop this function?

Reply:
You're right I'll take this out :)